lib/OpenQA/Worker/Settings.pm

```diff
@@ -19,17 +22,9 @@ has 'webui_host_specific_settings';
 use constant VNCPORT_OFFSET => $ENV{VNCPORT_OFFSET} // 90;
 
 sub new ($class, $instance_number = undef, $cli_options = {}) {
-    my $settings_file = ($ENV{OPENQA_CONFIG} || '/etc/openqa') . '/workers.ini';
-    my $cfg;
-    my @parse_errors;
-    if (-e $settings_file) {
-        $cfg = Config::IniFiles->new(-file => $settings_file);
-        push(@parse_errors, @Config::IniFiles::errors) unless $cfg;
-    }
-    else {
-        push(@parse_errors, "Config file not found at '$settings_file'.");
-        $settings_file = undef;
-    }
+    my $config_paths = lookup_config_files(path($Bin)->dirname->child('etc', 'openqa'), 'workers.ini', 1);
```

Comment:
Passing the home again here seems redundant? I guess I should have noticed this in the original PR. Shouldn't home be the same in all cases?

Reply:
Where is home passed again? What is passed here is the complete path to workers.ini

Reply:
The first argument is "home" in terms of lookup_config_files. Shouldn't the filename be enough when the API knows other locations, too?

Reply:
The `path($Bin)->dirname->child('etc', 'openqa')` part is redundant, indeed (as it is the same for the database initialization). I can improve that.

I guess name of the argument of `lookup_config_files` also shouldn't be called `$home` as it is actually a subdirectory within the `$home` directory which needs to be passed.

Reply:
Now I'm also wondering whether this `path($Bin)->dirname->child('etc', 'openqa')` approach is a good idea at all.

If we start packaging `/usr/etc/…` then the case of finding the config directory within the Git checkout will actually be accidentally "triggered" for a packaged installation (where binaries are under `/usr/bin/` and `/usr/bin/.../etc/` will be valid). Maybe this should use the home directory of the Mojo app in any case.